text/0083-sbt-standard.md

```diff
@@ -0,0 +1,252 @@
+
+# Specification
+
+SBT implements [NFT standard interface](https://github.com/ton-blockchain/TIPs/issues/62) but `transfer` should always be rejected, `pull_ownership` is used instead.
+
+#### 1. `pull_ownership`
+
+TL-B schema of inbound message:
+```
+pull_ownership#14726d45 query_id:uint64 signature:^(bits 512) 
+sbt_nonce:uint64 new_owner:MsgAddress response_destination:MsgAddress 
+custom_payload:(Maybe ^Cell) = InternalMsgBody;
+```
+`query_id` -  arbitrary request number.
+
+`signature` - signature of the rest part of the message.
+
+`sbt_nonce` - nonce, required for protection of signature replay attacks.
+
+`new_owner` - address of the new owner of the SBT item, should be the same as sender's address.
+
+`response_destination` - address where a response with confirmation of a successful pull and the rest of the incoming message coins should be sent.
+
+`custom_payload` - optional custom data.
+
+**Should be rejected if:**
+1. `signature` verification failed.
+2. `sbt_nonce` not equals stored nonce.
+3. `new_owner` is not sender's address.
+4. there is not enough coins (with respect to NFT own storage fee guidelines).
+5. After processing the request, the contract must send at least in_msg_value - forward_amount - max_tx_gas_price to the response_destination address. If the contract cannot guarantee this, it must immediately stop executing the request and throw the error.
+
+**Otherwise should do:**
+1. Change current owner of SBT to `new_owner` address.
+2. Generate and store new random `nonce`.
+3. Send all excesses of incoming message coins to response_destination with the following layout:
+   
+   `excesses#d53276db query_id:uint64 = InternalMsgBody;`
+
+   query_id should be equal with request's `query_id`.
+
+#### 2. `prove_ownership`
+
+TL-B schema of inbound message:
+```
+prove_ownership#5c9b0fb1 query_id:uint64 dest:MsgAddress 
+data:^Cell with_content:bool = InternalMsgBody;
+```
+`query_id` -  arbitrary request number.
+
+`dest` -  address of the contract to which the ownership of SBT should be proven.
+
+`data` - arbitrary data required by target contract.
+
+`with_content` - if true, SBT's content cell will be included in message to contract.
+
+**Should do:**
+
+Send message with TL-B schema to `dest` contract:
+```
+verify_ownership#a553079c query_id:uint64 sbt_id:uint256 owner:MsgAddress 
+data:^Cell content:(Maybe ^Cell) = InternalMsgBody;
+```
+
+`query_id` - request number passed in `prove_ownership`.
+
+`sbt_id` -  id of SBT.
+
+`owner` - current owner's address.
+
+`data` - data cell passed in `prove_ownership`.
+
+`content` - SBT's content, it is passed if `with_content` was true in `prove_ownership`.
+
+In case when `verify_ownership` was bounced back to SBT, SBT should send message to owner with schema:
+```
+verify_ownership_bounced#81b510c2 query_id:uint64 sbt_id:uint256 owner:MsgAddress 
```

Comment:
Does owner need sbt_id, owner, data and content in this case?

Reply:
It is like proxy of bounced message, technically i think most of that it is not so useful, but just for similarity with original.

As a use case, sbt_id can be used to confirm that bounce was from real SBT, by calculating address and comparing with sender.